### PR TITLE
Improve template capabilities of EMR job and step operators

### PR DESCRIPTION
**Pull request by oripwk**
The `EmrCreateJobFlowOperator` and `EmrAddStepsOperator` operators already support receiving `job_flow_overrides` and `steps` arguments as templates. However, one of the benefits of templating is the ability to refer to a file, which is not currently supported. A good example of an existing operator which leverages file templates is `BashOperator`, which applies this principle to its [bash_command](https://airflow.apache.org/docs/stable/_api/airflow/operators/bash_operator/index.html#airflow.operators.bash_operator.BashOperator) parameter.

Adding such support will improve the usability of the operators greatly, since at the moment the only way to do it is manually invoking Jinja to read the files before passing the arguments to the operators.

This PR simply adds the proper file extensions to `template_ext` field, and converts the resulting string into a Python dictionary.

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/emr_add_steps.py [hunk 1]
```diff
@@ -38,13 +38,14 @@ class EmrAddStepsOperator(BaseOperator):
...
-    :type steps: list
+    :param steps: boto3 style steps or reference to a steps file (must be '.json' or '.jinja2') to
+        be added to the jobflow. (templated)
+    :type steps: list|str
     :param do_xcom_push: if True, job_flow_id is pushed to XCom with key job_flow_id.
     :type do_xcom_push: bool
     """
     template_fields = ['job_flow_id', 'job_flow_name', 'cluster_states', 'steps']
-    template_ext = ()
+    template_ext = ('.json', '.jinja2')
```
Comment (ashb) [inline airflow/providers/amazon/aws/operators/emr_add_steps.py]: Nothing else uses `.jinja2` in Airflow, so I think we shouldn't create new pattern for just this operator.

Comment (oripwk) [reply airflow/providers/amazon/aws/operators/emr_add_steps.py]: I can remove it. Though it's a widespread convention, and also helps distinguish between real files and template files.

Comment (oripwk) [reply airflow/providers/amazon/aws/operators/emr_add_steps.py]: I now see that other operators with jsons such as `DruidOperator` and `SparkKubernetesOperator` don't use this. I'll update the PR to remove this.

Comment (oripwk) [reply airflow/providers/amazon/aws/operators/emr_add_steps.py]: @ashb Removed `.jinja2`

Comment (ashb) [reply airflow/providers/amazon/aws/operators/emr_add_steps.py]: I think this is a good idea (separating file from templated files) it's just not something we do else where yet, so we should talk about making this a bigger change afterwards

Comment (oripwk) [reply airflow/providers/amazon/aws/operators/emr_add_steps.py]: I think it's ready. Removed the `.jinja2` extension and also fixed a problem with the `template_ext` tuples.

**Review by ashb — APPROVED** [review]

**Pull request merged**